src/frontend/src/optimizer/plan_node/stream_hash_join.rs

```diff
@@ -181,46 +175,48 @@ impl_plan_tree_node_for_binary! { StreamHashJoin }
 
 impl ToStreamProst for StreamHashJoin {
     fn to_stream_prost_body(&self) -> NodeBody {
+        let left_key_indices_prost = self
+            .eq_join_predicate
+            .left_eq_indexes()
+            .iter()
+            .map(|v| *v as i32)
+            .collect_vec();
+        let right_key_indices_prost = self
+            .eq_join_predicate
+            .right_eq_indexes()
+            .iter()
+            .map(|v| *v as i32)
+            .collect_vec();
+        let left_key_indices = left_key_indices_prost
+            .iter()
+            .map(|idx| *idx as usize)
+            .collect_vec();
+        let right_key_indices = right_key_indices_prost
+            .iter()
+            .map(|idx| *idx as usize)
+            .collect_vec();
         NodeBody::HashJoin(HashJoinNode {
             join_type: self.logical.join_type() as i32,
-            left_key: self
-                .eq_join_predicate
-                .left_eq_indexes()
-                .iter()
-                .map(|v| *v as i32)
-                .collect(),
-            right_key: self
-                .eq_join_predicate
-                .right_eq_indexes()
-                .iter()
-                .map(|v| *v as i32)
-                .collect(),
+            left_key: left_key_indices_prost,
+            right_key: right_key_indices_prost,
             condition: self
                 .eq_join_predicate
                 .other_cond()
                 .as_expr_unless_true()
                 .map(|x| x.to_expr_proto()),
-            dist_key_l: self
-                .dist_key_l
-                .dist_column_indices()
-                .iter()
-                .map(|idx| *idx as u32)
-                .collect_vec(),
-            dist_key_r: self
-                .dist_key_r
-                .dist_column_indices()
-                .iter()
-                .map(|idx| *idx as u32)
-                .collect_vec(),
             is_delta_join: self.is_delta,
-            left_table: Some(infer_internal_table_catalog(self.left()).to_prost(
-                SchemaId::placeholder() as u32,
-                DatabaseId::placeholder() as u32,
-            )),
-            right_table: Some(infer_internal_table_catalog(self.right()).to_prost(
-                SchemaId::placeholder() as u32,
-                DatabaseId::placeholder() as u32,
-            )),
+            left_table: Some(
+                infer_internal_table_catalog(self.left(), left_key_indices).to_prost(
+                    SchemaId::placeholder() as u32,
+                    DatabaseId::placeholder() as u32,
+                ),
+            ),
```

Comment:
What about impl infer_internal_table_catalog as a method in LogicalJoin?

Reply:
I think it should remain in StreamHashJoin since it is only used in streaming.